### PR TITLE
fix(40590): bundle the readme files of datasources inside app plugins

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What this PR does / why we need it**:
Currently we only bundle the root README.md file into the build of a plugin. This PR also bundles all the README files in subfolders.

**Which issue(s) this PR fixes**:

Fixes #40590

**Motivation behind this PR**
- Mostly this one is a discussion point for us to figure out if we want to keep the readme's in the plugin package, not necessary something to merge
- It seems that current version of Grafana does not care about the nested Readme's so there might be no benefit in doing so from the displaying perspective at the moment, the question is more - why strip it away from the datasources when it's nested inside of an app?
- Relevant code is [here](https://github.com/grafana/grafana/blob/main/public/app/features/plugins/admin/api.ts#L74) and the backend side is [here](https://github.com/grafana/grafana/blob/main/pkg/api/plugins.go#L209)
- It also seems that older version included the nested readmes in the Tab called '`README`' (see screenshots, but not 100% sure about it)
image (1)
image

### CODE REVIEW

**Review by leventebalogh — COMMENTED** [review]
I guess the goal is to be able to display `README`s for nested datasource plugins inside the plugins catalog, am I right?

Did you try out if it works as intended if we build a plugin with nested datasources? I am just a bit concerned because of how we are locating the README files. 

The website should keep working fine, as it is looking for the README file next to the main `plugin.json`. ([source](https://github.com/grafana/grafana-com/blob/cc3660932cd952789fcd596433b6fb6b5c268a49/api/lib/plugin-utils.js#L370)).

I will try it out locally as well and get back.

**Comment by tolzhabayev** [comment]
> I guess the goal is to be able to display `README`s for nested datasource plugins inside the plugins catalog, am I right?
> 
> Did you try out if it works as intended if we build a plugin with nested datasources? I am just a bit concerned because of how we are locating the README files.
> 
> The website should keep working fine, as it is looking for the README file next to the main `plugin.json`. ([source](https://github.com/grafana/grafana-com/blob/cc3660932cd952789fcd596433b6fb6b5c268a49/api/lib/plugin-utils.js#L370)).
> 
> I will try it out locally as well and get back.



> I guess the goal is to be able to display `README`s for nested datasource plugins inside the plugins catalog, am I right?
> 
> Did you try out if it works as intended if we build a plugin with nested datasources? I am just a bit concerned because of how we are locating the README files.
> 
> The website should keep working fine, as it is looking for the README file next to the main `plugin.json`. ([source](https://github.com/grafana/grafana-com/blob/cc3660932cd952789fcd596433b6fb6b5c268a49/api/lib/plugin-utils.js#L370)).
> 
> I will try it out locally as well and get back.

Added more context into the description. This is more of a `discussion PR` - so for us to see if we want README's bundled in or not. Not necessarily to merge.

File: packages/grafana-toolkit/src/config/webpack.plugin.config.ts [hunk 1]
```diff
@@ -99,6 +99,7 @@ const getCommonPlugins = (options: WebpackConfigurationOptions) => {
         // If src/README.md exists use it; otherwise the root README
         { from: hasREADME ? 'README.md' : '../README.md', to: '.', force: true },
         { from: 'plugin.json', to: '.' },
+        { from: '**/README.md', to: '[path]README.md' },
```
Comment (jackw) [inline packages/grafana-toolkit/src/config/webpack.plugin.config.ts:102]: Do we need to make use of [`priority`](https://webpack.js.org/plugins/copy-webpack-plugin/#priority) so we can control how it plays with the `README.md` copy happening above on line 100?

Comment (tolzhabayev) [reply packages/grafana-toolkit/src/config/webpack.plugin.config.ts]: @jackw good point https://github.com/grafana/grafana/pull/44068/commits/7d0b2208fbba429779ee10e86f7af1f91e0f1e50

**Review by mckn — COMMENTED** [review]
I think this is great! My only concern is regarding security and a risk of including readme files in the bundle that should not be included. But not sure it is a valid concern.

**Review by mckn — APPROVED** [review]
LGTM

**Review by leventebalogh — APPROVED** [review]

**Pull request merged**